lib/dpl/provider/pages.rb

```diff
@@ -44,34 +53,140 @@ def slug
         options.fetch(:repo) { context.env['TRAVIS_REPO_SLUG'] }
       end
 
+      def keep_history
+        options.fetch(:keep_history) { false }
+      end
+
+      def allow_empty_commit
+        options.fetch(:allow_empty_commit) { false }
+      end
+
+      def verbose
+        # Achtung! Never verbosify git, since it may expose user's token.
+        options.fetch(:verbose) { false }
+      end
+
       def check_auth
       end
 
       def needs_key?
         false
       end
 
-      def github_deploy
-        context.shell 'rm -rf .git > /dev/null 2>&1'
-        context.shell "touch \"deployed at `date` by #{@gh_name}\""
-        context.shell 'git init' or raise 'Could not create new git repo'
+      def print_step(msg)
+        $stderr.puts msg if @verbose
```

Comment:
There's a `log` member in https://github.com/travis-ci/dpl/blob/6c3fddfda1f2a85944c544446b068bac0a77c049/lib/dpl/provider.rb#L248 that you should probably use instead of `$stderr.puts`.